src/OrchardCore.Modules/OrchardCore.OpenId/Recipes/OpenIdApplicationStep.cs

```diff
@@ -29,24 +29,65 @@ public async Task ExecuteAsync(RecipeExecutionContext context)
                 return;
             }
 
-            var model = context.Step.ToObject<CreateOpenIdApplicationViewModel>();
+            var model = context.Step.ToObject<OpenIdApplicationStepModel>();
+            var application = await _applicationManager.FindByClientIdAsync(model.ClientId);
 
-            var descriptor = new OpenIdApplicationDescriptor
+            var descriptor = new OpenIdApplicationDescriptor();
+            if (application!=null)
             {
-                ClientId = model.ClientId,
-                ClientSecret = model.ClientSecret,
-                ConsentType = model.ConsentType,
-                DisplayName = model.DisplayName,
-                Type = model.Type
-            };
+                await _applicationManager.PopulateAsync(application, descriptor);
 
+                descriptor.Permissions.Remove(OpenIddictConstants.Permissions.GrantTypes.ClientCredentials);
```

Comment:
It would have been better to use the same `if`/`else` logic as the one used in the controllers to make things more consistent and avoid discrepancies between recipes and controllers, but it shouldn't be blocking to merge this PR,

Reply:
Is it better to create an OpenIdApplicationService with an Create and Edit function with an ApplicationModel?
Then use this service in the recipe and the controller?

Reply:
It's an option, but the problem is that the step and view models are 2 different CLR classes, so we'd to either merge the classes or introduce some interfaces. Dunno if it's worth it or not.

If anything, it would be better to put that in `IOpenIdApplicationManager`.